beluga/test/beluga/views/test_random_intersperse.cpp

```diff
@@ -0,0 +1,142 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include "beluga/views/random_intersperse.hpp"
+
+#include <range/v3/algorithm/count.hpp>
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/generate.hpp>
+#include <range/v3/view/iota.hpp>
+#include <range/v3/view/take.hpp>
+
+namespace {
+
+TEST(RandomIntersperseView, ConceptChecksFromContiguousRange) {
+  auto input = std::array{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+  auto output = beluga::views::random_intersperse(input, []() { return 0; });
+
+  static_assert(ranges::common_range<decltype(input)>);
+  static_assert(!ranges::common_range<decltype(output)>);
+
+  static_assert(!ranges::viewable_range<decltype(input)>);
+  static_assert(ranges::viewable_range<decltype(output)>);
+
+  static_assert(ranges::forward_range<decltype(input)>);
+  static_assert(ranges::forward_range<decltype(output)>);
+
+  static_assert(ranges::sized_range<decltype(input)>);
+  static_assert(!ranges::sized_range<decltype(output)>);
+
+  static_assert(ranges::bidirectional_range<decltype(input)>);
+  static_assert(!ranges::bidirectional_range<decltype(output)>);
+
+  static_assert(ranges::random_access_range<decltype(input)>);
+  static_assert(!ranges::random_access_range<decltype(output)>);
+
+  static_assert(ranges::contiguous_range<decltype(input)>);
+  static_assert(!ranges::contiguous_range<decltype(output)>);
+}
+
+TEST(RandomIntersperseView, ConceptChecksFromInfiniteRange) {
+  auto input = ranges::views::generate([]() { return 1; });
+  auto output = beluga::views::random_intersperse(input, []() { return 0; });
+
+  static_assert(!ranges::common_range<decltype(input)>);
+  static_assert(!ranges::common_range<decltype(output)>);
+
+  static_assert(ranges::viewable_range<decltype(input)>);
+  static_assert(ranges::viewable_range<decltype(output)>);
+
+  static_assert(!ranges::forward_range<decltype(input)>);
+  static_assert(!ranges::forward_range<decltype(output)>);
+
+  static_assert(!ranges::sized_range<decltype(input)>);
+  static_assert(!ranges::sized_range<decltype(output)>);
+
+  static_assert(!ranges::bidirectional_range<decltype(input)>);
+  static_assert(!ranges::bidirectional_range<decltype(output)>);
+
+  static_assert(!ranges::random_access_range<decltype(input)>);
+  static_assert(!ranges::random_access_range<decltype(output)>);
+
+  static_assert(!ranges::contiguous_range<decltype(input)>);
+  static_assert(!ranges::contiguous_range<decltype(output)>);
+}
+
+TEST(RandomIntersperseView, GuaranteedIntersperseFirstElement) {
+  auto input = std::array{10, 20, 30};
+  auto output = input | beluga::views::random_intersperse([i = 0]() mutable { return i++; }, 1.0);
```

Comment:
@nahueespinosa nit: consider using a named constant for the intersperse probability, here and elsewhere.

Reply:
Done in https://github.com/Ekumen-OS/beluga/pull/278/commits/ec93c23f1f4c3c0ee429bfe383df45508cf744ea.